libraries/AP_CANManager/AP_CANManager.cpp

```diff
@@ -318,15 +320,13 @@ void AP_CANManager::log_text(AP_CANManager::LogLevel loglevel, const char *tag,
 }
 
 // log retrieve method used by file sys method to report can log
-uint32_t AP_CANManager::log_retrieve(char* data, uint32_t max_size) const
+void AP_CANManager::log_retrieve(ExpandingString *str) const
 {
     if (_log_buf == nullptr) {
         gcs().send_text(MAV_SEVERITY_ERROR, "Log buffer not available");
-        return 0;
+        return;
     }
-    uint32_t read_len = MIN(max_size, _log_pos);
-    memcpy(data, _log_buf, read_len);
-    return read_len;
+    str->append(_log_buf, _log_pos);
```

Comment:
Given these semantics at the very least the comment should mention that the string is *appended* to `str` - my default assumption based on the function name would be that `str` would be set to the log text.

Reply:
the ExpandingString method does an append. It just so happens this use case only calls it once. So both names are correct